ReSharper.FSharp/src/FSharp.Psi.Intentions/src/QuickFixes/ReplaceReturnTypeFix.fs

```diff
@@ -0,0 +1,85 @@
+namespace JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.QuickFixes
+
+open System.Text.RegularExpressions
+open JetBrains.ReSharper.Psi.ExtensionsAPI.Tree
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Features.Daemon.Highlightings
+open JetBrains.ReSharper.Plugins.FSharp.Psi.Tree
+open JetBrains.ReSharper.Psi.ExtensionsAPI
+open JetBrains.ReSharper.Resources.Shell
+
+type ReplaceReturnTypeFix(expr: IFSharpExpression, fcsErrorMessage: string, replaceFirstType: bool) =
+    inherit FSharpQuickFixBase()
+    
+    let rec visitParent (expr: IFSharpExpression): IFSharpExpression option =
+        match expr.Parent with
+        | :? ISequentialExpr as seqExpr ->
+            if seqExpr.Expressions.Last() <> expr then None else
+            visitParent seqExpr
+        | :? IMatchClause as clause ->
```

Comment:
Getting the match clause parent like this is not good here. Is it the `when` expression or is it the expression after `->`? We're only interested in the last one here.
There're generated methods that help with such navigation, take a look at `MatchClauseNavigator` and similar classes.